lib/features/base/base_controller.dart

```diff
@@ -119,71 +153,104 @@ abstract class BaseController extends GetxController
+    if (PlatformInfo.isWeb) {
+      handleUrgentExceptionOnWeb(failure: failure, exception: exception);
+    } else if (PlatformInfo.isMobile) {
+      handleUrgentExceptionOnMobile(failure: failure, exception: exception);
+    } else {
+      throw NoSupportPlatformException();
+    }
+  }
+
+  void handleUrgentExceptionOnMobile({Failure? failure, Exception? exception}) {
+    logError('$runtimeType::handleUrgentExceptionOnMobile():Failure: $failure | Exception: $exception');
     if (exception is ConnectionError) {
-      if (currentOverlayContext != null && currentContext != null) {
-        appToast.showToastErrorMessage(
-          currentOverlayContext!,
-          AppLocalizations.of(currentContext!).connectionError
-        );
-      }
+      _handleConnectionErrorException();
+    } else if (exception is BadCredentialsException) {
+      _handleBadCredentialsException();
     }
+  }
 
-    if (!authorizationInterceptors.isAppRunning) {
-      return;
+  void handleUrgentExceptionOnWeb({Failure? failure, Exception? exception}) {
+    logError('$runtimeType::handleUrgentExceptionOnWeb():Failure: $failure | Exception: $exception');
+    if (exception is NoNetworkError) {
+      _handleNotNetworkErrorException();
+    } else if (exception is ConnectionError) {
+      _handleConnectionErrorException();
+    } else if (exception is BadCredentialsException) {
+      _handleBadCredentialsException();
     }
-    _executeBeforeUnloadAndLogOut(exception);
   }
 
-  Future<void> _executeBeforeUnloadAndLogOut(Exception? exception) async {
-    if (exception is BadCredentialsException || exception is ConnectionError) {
-      if (PlatformInfo.isWeb) {
-        await executeBeforeUnload();
+  Future<void> _executeBeforeUnloadAndLogOut() async {
+    await executeBeforeUnload();
+    clearDataAndGoToLoginPage();
+  }
+
+  void _handleConnectionErrorException() {
+    if (currentOverlayContext != null && currentContext != null) {
+      appToast.showToastErrorMessage(
+        currentOverlayContext!,
+        AppLocalizations.of(currentContext!).connectionError);
+    }
+  }
+
+  void _handleNotNetworkErrorException() {
+    if (currentOverlayContext != null && currentContext != null) {
+      appToast.showToastMessage(
+        currentOverlayContext!,
+        AppLocalizations.of(currentContext!).no_internet_connection,
+        actionName: AppLocalizations.of(currentContext!).skip,
+        onActionClick: ToastView.dismiss,
+        leadingSVGIcon: imagePaths.icNotConnection,
+        backgroundColor: AppColor.textFieldErrorBorderColor,
+        textColor: Colors.white,
+        infinityToast: true);
+    }
+  }
+
+  void _handleBadCredentialsException() {
+    if (PlatformInfo.isWeb) {
+      if (currentContext == null) {
+        _executeBeforeUnloadAndLogOut();
+        return;
+      }
+
+      bool isConfirm = html.window.confirm(
+        AppLocalizations.of(currentContext!).dialogMessageSessionHasExpired
+      );
```

Comment:
we dont need to `await` here?

Reply:
No, we will remove it and use dialog flutter